bin/generate-bundle.js

```diff
@@ -0,0 +1,233 @@
+        console.log(`Modules processed: ${processed.join(',')} for ${dsV}`)
+
+        console.log(`Skipped ${skipped.length} modules for ${dsV}`)
+        console.log(`Modules skipped: ${skipped.join(',')} for ${dsV}`)
+      }
+      resolve(compiled);
+    });
+  }
+
+  generateLESS() {
+    return this.getDistCss(this.dsV).then((files) => this.processFiles(files));
+  }
+
+  compileLess(raw, plugin) {
+    return less.render(raw, { plugins: [plugin] })
+      .then((render) => render.css,
+      );
+  }
+
+  writeAllFiles(raw) {
+    const path = getCDNPath(this.args.name, this.dsV);
+    return makeDir(path).then(() => writeFile(`${path}/skin.min.css`, raw));
+  }
+}
+
+function getCDNPath(bundle, dsV) {
+  return `${currentDir}/_cdn/${bundle}/v${pkg.version}/${dsV}`;
+}
+
+
+/**
+ * Runs the compilation of less
+ * @param {*} res The response with dsVersion and raw css/less
+ * @param {*} plugin The given plugin to run the render with
+ */
+/**
+ * Runs npm build to get dist output
+ */
+function prebuild() {
+  return new Promise((resolve, reject) => {
+    console.log('Running build...')
+    exec('npm run build', (err) => {
+      if (err) {
+        return reject(err);
+      }
+      console.log('npm build successful')
+      return resolve();
+    });
+  });
+}
+
+function makeDir(path) {
+  return new Promise((resolve, reject) => {
+    fs.mkdir(path, { recursive: true }, (err) => {
+      if (err) {
+        return reject(err);
+      }
+      resolve();
+    });
+  });
+}
+
+function writeFile(file, data) {
+  return new Promise((resolve, reject) => {
+    fs.writeFile(file, data, (err) => {
+      if (err) {
+        return reject(err);
+      }
+      resolve();
+    });
+  });
+}
+
+
+
+/**
+ * Main function to start, generates less files and compiles and writes them
+ * @param {*} name
+ * @param {*} args
+ */
+function runCSSBuild(name, args) {
+  prebuild().then(() => Promise.all(
+    dsList.map((ds) => {
+      const cssProcesser = new CssProcesser(ds, args);
+      return cssProcesser.run();
+    })))
+    .then(() => {
+      console.log(`Bundles created successfully!
+Please upload the ./_cdn/${args.name}/${pkg.version} directory to CDN
+`);
+    }).catch((e) => {
+      console.error(e);
+    });
+}
+
+
+require('yargs') // eslint-disable-line
+  .command('list', 'List all available modules', (yargs) => {
+  }, (argv) => {
+    prebuild().then(() => {
+      dsList.forEach((dsV) => {
+        const cssProcesser = new CssProcesser(dsV, argv);
+        cssProcesser.getDistCss().then((files) => {
+          console.log(`======================`);
+          console.log(`${dsV} - modules avaiable`);
+          console.log(`======================`);
+          files.forEach((file) => {
+            console.log(path.basename(file, '.css'));
+          })
+          console.log()
+          console.log()
+
+        })
+      })
+    });
+  })
+  .command('gen [name]', 'generates a CDN bundle with the given name', (yargs) => {
```

Comment:
Could we make this `generate`? Makes it clear what it does. I like shorthand, but being explicit helps.

Alternatives are `build`, `create`, `bundle`, `pack`, etc.

Reply:
Changed to bundle